platform/view/services/db/db.go

```diff
@@ -70,14 +74,14 @@ func Open(driverName, dataSourceName string) (driver.Persistence, error) {
 // driverName is a string that describes the driver
 // dataSourceName describes the data source in a driver-specific format.
 // The returned connection is only used by one goroutine at a time.
-func OpenVersioned(driverName, dataSourceName string) (driver.VersionedPersistence, error) {
+func OpenVersioned(sp view2.ServiceProvider, driverName, dataSourceName string, config Config) (driver.VersionedPersistence, error) {
```

Comment:
This API change seems to be only needed for the case when we want to use Orion as storage backend. All other implementations do not need the SP, right?

Would be nice to have solve this differently. Can't we use some factory pattern triggered from the sdk.go that checks what driver is selected, and in case of orion, we pass the sp to some orion factory and then wrap the orion instance behind this db abstraction?!

Reply:
Not sure I get this but we can certainly address this refactoring in a dedicated PR where we expend the interface to support concurrent transactions. Wdyt?